next.config.js

```diff
@@ -41,7 +41,21 @@ const nextConfig = {
       },
     ];
   },
+  async rewrites() {
+    return [
+      // planet APi to get tiles
+      {
+        source: '/planet/:path*',
+        destination: `https://tiles.planet.com/basemaps/v1/planet-tiles/:path*?api_key=${process.env.NEXT_PUBLIC_PLANET_API_KEY}`,
+      },
 
+      // Planet API get mosaics
+      {
+        source: '/planet-api/:path*',
+        destination: `https://api.planet.com/basemaps/v1/:path*?api_key=${process.env.NEXT_PUBLIC_PLANET_API_KEY}`,
+      },
+    ];
+  },
```

Comment:
You can add the API key on fly when the map is about to request the tile: https://github.com/resource-watch/resource-watch/blob/develop/components/map/component.tsx#L351-L367

Reply:
What's wrong with this solution? or if just a "sharing knowledge" moment?

Reply:
Nothing wrong, and it probably works as expected. But the map already provides a way to handle this specific case without messing with the global Next configuration. If something is wrong with the application in the near future, debugging it with an additional configuration like this makes things harder. The cleaner we can keep the configuration, the more resilient the application will be.

Of course, there are cases you have no alternative, like setting up redirects, and that's fine. But for this specific case, the library `react-map-gl` can handle it.

Reply:
I have an API call as well, not only the tiles